network/auth-header/src/index.ts

```diff
@@ -38,8 +38,18 @@ function getAuthHeaderByURI (authHeaders: Record<string, string>, maxParts: numb
 }
 
 function removePort (originalUrl: string) {
+  const hasHTTPS = originalUrl.match(':443')
   const urlObj = new URL(originalUrl)
+
+  /**
+   * @artifactory adds a https port in the middle of their urls and URL.parse will not return ports in that range
+   */
+  if (hasHTTPS && urlObj.port === '') {
+    return urlObj.href
+  }
+
   if (urlObj.port === '') return originalUrl
```

Comment:
It is probably just enough to do this:
```suggestion
  if (urlObj.port === '') return urlObj.href
```

it will handle both :80 and :443

Reply:
@zkochan agreed. I added tests for `http|https|ws|wss` and moved the `removePort` method to it's own file so it can be tested without exporting it.